packages/next-auth/src/providers/strava.ts

```diff
@@ -0,0 +1,60 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+export interface StravaProfile extends Record<string, any> {
+  id: string // this is really a number
+  username: string
+  resource_state: number
+  firstname: string
+  lastname: string
+  bio: string
+  city: string
+  state?: any
+  country?: any
+  sex: string
+  premium: boolean
+  summit: boolean
+  created_at: string
+  updated_at: string
+  badge_type_id: number
+  weight: number
+  profile_medium: string
+  profile: string
+  friend?: any
+  follower?: any
+}
+
+export default function Strava<P extends StravaProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "strava",
+    name: "Strava",
+    type: "oauth",
+    authorization: {
+      url: "https://www.strava.com/api/v3/oauth/authorize",
+      params: {
+        scope: "read",
+        approval_prompt: "auto",
+        response_type: "code",
+        redirect_uri: `${process.env.NEXTAUTH_URL|| `https:localhost:3000`}/api/auth/callback/strava`,
```

Comment:
Not sure if using `process.env.NEXTAUTH_URL` is the convention, but it solved my issue. I presume it would cover 90% of use cases. Not sure whether to include `https://` in `redirect_uri` URL construction.